tests/test_enum.py

```diff
@@ -1,7 +1,44 @@
 # -*- coding: utf-8 -*-
 import pytest
+
+import env  # noqa: F401
+
 from pybind11_tests import enums as m
 
+if env.PY2:
+    enum = None
+else:
+    import enum
+
+
+def is_enum(cls):
+    """Example showing how to recognize a class as pybind11 enum or a
+    PEP 345 enum."""
+    if enum is not None:
+        if issubclass(cls, enum.Enum):
+            return True
+    return getattr(cls, "is_pybind11_enum", False)
+
+
+def test_pep435():
+    # See #2332.
+    cls = m.UnscopedEnum
+    names = ("EOne", "ETwo", "EThree")
+    values = (cls.EOne, cls.ETwo, cls.EThree)
+    raw_values = (1, 2, 3)
+
+    assert len(cls) == len(names)
+    assert list(cls) == list(values)
+    assert is_enum(cls)
+    if enum:
+        assert not issubclass(cls, enum.Enum)
+    for name, value, raw_value in zip(names, values, raw_values):
+        assert isinstance(value, cls)
+        if enum:
+            assert not isinstance(value, enum.Enum)
+        assert value.name == name
+        assert value.value == raw_value
```

Comment:
Self-note: Should try to implement `Flag` based interface. (i.e. closure under arithmetic operations)